metricq/agent.py

```diff
@@ -184,6 +184,17 @@ def __init__(
             __version__,
         )
 
+    @property
+    def url(self) -> str:
+        """
+        Redacts login information from the management URL.
+        """
+        return (
+            str(URL(self._management_url).with_password("******"))
+            if URL(self._management_url).user
+            else self._management_url
+        )
```

Comment:
It's not wrong, but it feels wrong. Why use an if-else expression and then stretch it on multiple lines?
The normal if-statement is much more readable here.

Also, parsing the URL string into a URL instance certainly isn't cheap.

```suggestion
        url = URL(self._management_url)
        if url.user:
            return str(url.with_password("******"))
        return self._management_url
```